modules/contentpool_remote_register/src/Plugin/rest/resource/ContentpoolTermReferenceFieldsResource.php

```diff
@@ -0,0 +1,157 @@
+   *   The plugin definition.
+   * @param array $serializer_formats
+   *   The array of serializer formats.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   The logger service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   * @param \Drupal\Core\Entity\EntityFieldManagerInterface $entity_field_manager
+   *   The entity field manager.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, array $serializer_formats, LoggerInterface $logger, EntityTypeManagerInterface $entity_type_manager, EntityFieldManagerInterface $entity_field_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $serializer_formats, $logger);
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityFieldManager = $entity_field_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->getParameter('serializer.formats'),
+      $container->get('logger.factory')->get('rest'),
+      $container->get('entity_type.manager'),
+      $container->get('entity_field.manager')
+    );
+  }
+
+  /**
+   * Provides a response for the endpoint.
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   The request object.
+   *
+   * @return \Drupal\rest\ModifiedResourceResponse
+   *   Response.
+   */
+  public function get(Request $request) {
+    $entity_type = $request->query->get('entity_type');
+    $bundle = $request->query->get('bundle');
+
+    $response_data = [];
+
+    $field_definitions = $this->entityFieldManager->getFieldDefinitions($entity_type, $bundle);
+    /** @var \Drupal\Core\Field\FieldDefinitionInterface[] $term_reference_fields */
+    $term_reference_fields = array_filter($field_definitions, function (FieldDefinitionInterface $field_definition) {
+      /** @var \Drupal\field\FieldStorageConfigInterface $storage_config */
+      $storage_config = $field_definition->getFieldStorageDefinition();
+      return $storage_config->getType() == 'entity_reference' && $storage_config->getSetting('target_type') == 'taxonomy_term';
+    });
+
+    /** @var \Drupal\taxonomy\TermStorageInterface $term_storage */
+    $term_storage = $this->entityTypeManager->getStorage('taxonomy_term');
+
+    foreach ($term_reference_fields as $field_definition) {
+      $taxonomy_id = reset($field_definition->getSetting('handler_settings')['target_bundles']);
+      $term_tree = $term_storage->loadTree($taxonomy_id, 0, NULL, TRUE);
+      $response_data[$field_definition->getName()] = [
+        'id' => $taxonomy_id,
+        'field' => $field_definition->getName(),
+        'label' => $field_definition->getLabel(),
+        'terms' => static::buildResponseTree($term_tree),
```

Comment:
shouldn't be static when there isn't really a reason to?

Reply:
js check in phpstorm always says method should be static if there is no this-reference.

Reply:
dont know about php though, but its protected anyways.

Reply:
that js sounds stupid. it should be $this-> and non-static, but minor.